wasm/CALLER.wast

```diff
@@ -1,23 +1,12 @@
 (func $CALLER
   (param $sp i32)
   (result i32)
-
-  (local $scratch i32)
-
   ;; there's no input item for us to overwrite
   (set_local $sp (i32.add (get_local $sp) (i32.const 32)))
 
-  (set_local $scratch (i32.const 32776))
-
-  ;; clear the 3rd 64bit chunk so we don't need to mask it later
-  (i64.store (i32.add (get_local $scratch) (i32.const 16)) (i64.const 0))
-
   ;; loads the caller into memory
-  (call_import $caller (get_local $scratch))
-
-  (i64.store (i32.add (get_local $sp) (i32.const 0)) (i64.load (i32.add (get_local $scratch) (i32.const 0))))
-  (i64.store (i32.add (get_local $sp) (i32.const 8)) (i64.load (i32.add (get_local $scratch) (i32.const 8))))
-  (i64.store (i32.add (get_local $sp) (i32.const 16)) (i64.load (i32.add (get_local $scratch) (i32.const 16))))
+  (call_import $caller (get_local $sp))
+  ;; zero out the last 64 bits
   (i64.store (i32.add (get_local $sp) (i32.const 24)) (i64.const 0))
```

Comment:
This is invalid. The address is 160 bits and this only masks the top 64 bits of the 256 bits. There's 32 bits of uncertain data.